client/src/error.rs

```diff
@@ -205,47 +201,9 @@ pub enum Error {
     /// URL validation error
     #[error("{0}")]
     UrlValidation(String),
-
-    //////////////////////////////////////////////////////////////////////
-    // Input Selection
-    //////////////////////////////////////////////////////////////////////
-    /// Required input is forbidden.
-    #[error("required input {0} is forbidden")]
-    RequiredInputIsForbidden(OutputId),
-    /// Required input is not available.
-    #[error("required input {0} is not available")]
-    RequiredInputIsNotAvailable(OutputId),
-    /// Unfulfillable requirement.
-    #[error("unfulfillable requirement {0:?}")]
-    // TODO better name?
-    UnfulfillableRequirement(Requirement),
-    /// No available inputs were provided to input selection
-    #[error("no available inputs provided")]
-    NoAvailableInputsProvided,
-    /// No outputs were provided to input selection
-    #[error("no outputs provided")]
-    NoOutputsProvided,
-    /// Insufficient amount provided.
-    #[error("insufficient amount: found {found}, required {required}")]
-    InsufficientAmount {
-        /// The amount found.
-        found: u64,
-        /// The required amount.
-        required: u64,
-    },
-    /// Insufficient native token amount provided.
-    #[error("insufficient native token amount: found {found}, required {required}")]
-    InsufficientNativeTokenAmount {
-        /// The token ID.
-        token_id: TokenId,
-        /// The amount found.
-        found: U256,
-        /// The required amount.
-        required: U256,
-    },
-    /// Can't burn and transition an output at the same time.
-    #[error("can't burn and transition an output at the same time, chain ID: {0}")]
-    BurnAndTransition(ChainId),
+    /// Input selection error.
+    #[error("{0}")]
+    InputSelection(#[from] InputSelectionError),
 
```

Comment:
What about the following variants? The are not in the ISA core, but still related to input selection I would say
````Rust
    /// The wallet account has enough funds, but split on too many outputs
    #[error("the wallet account has enough funds, but split on too many outputs: {0}, max. is 128, consolidate them")]
    ConsolidationRequired(usize),
    /// Missing input for utxo chain
    #[error("missing input: {0}")]
    MissingInput(String),
    /// No input with matching ed25519 address provided
    #[error("no input with matching ed25519 address provided")]
    MissingInputWithEd25519Address,
    /// The wallet account doesn't have any inputs found
    #[error("no inputs found")]
    NoInputs,
    /// The wallet account doesn't have enough native tokens
    #[error("the wallet account doesn't have enough native tokens, missing: {0:?}")]
    NotEnoughNativeTokens(NativeTokens),
    /// The wallet account doesn't have enough balance for an output with the remaining native tokens.
    #[error("the wallet account doesn't have enough balance for an output with the remaining native tokens.")]
    NoBalanceForNativeTokenRemainder,
```
Some can also be removed or changed I think, for example `ConsolidationRequired` is more like `TooManyInputs`, `NotEnoughNativeTokens` is basically the same as `InsufficientNativeTokenAmount` and `NoBalanceForNativeTokenRemainder` is just a clearer name for `InsufficientAmount`
`MissingInputWithEd25519Address` is in the new Error, but not removed here

Reply:
`MissingInput` was not used anymore so I removed it.
`MissingInputWithEd25519Address` removed from the main error type.
`NoInputs` replaced by existing `NoAvailableInputsProvided`.
`NoBalanceForNativeTokenRemainder` wasn't used anymore, removed.
`NotEnoughNativeTokens` wasn't used anymore, removed.
`ConsolidationRequired` I personally wouldn't put in ISA tbh, or later when I'm done with other error refactoring see where it best fits.